[STM32F7xx] update cube hal to v1.4, adding NUCLEO_F767

ohagendorf opened the pull request:
test results of existing targets are the same - please see the new results below
test results of new NUCLEO_F767 are OK - please see the results below
both NUCLEO_F7x are added to travis build

Result  | Target        | Toolchain | Test ID     | Test Description                      | Elapsed Time (sec) | Timeout (sec) | Loops
---------|---------------|-----------|-------------|---------------------------------------|--------------------|---------------|-------
OK      | NUCLEO_F746ZG | GCC_ARM   | DTCT_1      | Simple detect test                    |        0.5                |10      |  1/1 
OK      | NUCLEO_F746ZG | GCC_ARM   | EXAMPLE_1   | /dev/null                             |        3.43        |       20      |  1/1 
OK      | NUCLEO_F746ZG | GCC_ARM   | MBED_10     | Hello World                           |        0.36        |       5       |  1/1 
OK      | NUCLEO_F746ZG | GCC_ARM   | MBED_11     | Ticker Int                            |       11.37        |       15      |  1/1 
OK      | NUCLEO_F746ZG | GCC_ARM   | MBED_12     | C++                                   |        1.37        |       10      |  1/1 
OK      | NUCLEO_F746ZG | GCC_ARM   | MBED_16     | RTC                                   |        4.59        |       20      |  1/1 
OK      | NUCLEO_F746ZG | GCC_ARM   | MBED_2      | stdio                                 |        0.77        |       20      |  1/1 
OK      | NUCLEO_F746ZG | GCC_ARM   | MBED_23     | Ticker Int us                         |       11.36        |       15      |  1/1 
OK      | NUCLEO_F746ZG | GCC_ARM   | MBED_24     | Timeout Int us                        |       11.41        |       15      |  1/1 
OK      | NUCLEO_F746ZG | GCC_ARM   | MBED_25     | Time us                               |       11.38        |       15      |  1/1 
OK      | NUCLEO_F746ZG | GCC_ARM   | MBED_26     | Integer constant division             |        1.39        |       20      |  1/1 
OK      | NUCLEO_F746ZG | GCC_ARM   | MBED_34     | Ticker Two callbacks                  |       11.41        |       15      |  1/1 
OK      | NUCLEO_F746ZG | GCC_ARM   | MBED_37     | Serial NC RX                          |       10.91        |       20      |  1/1  
OK      | NUCLEO_F746ZG | GCC_ARM   | MBED_38     | Serial NC TX                          |        15.4        |       20      |  1/1 
OK      | NUCLEO_F746ZG | GCC_ARM   | MBED_A1     | Basic                                 |        1.35        |       20      |  1/1 
OK      | NUCLEO_F746ZG | GCC_ARM   | MBED_A21    | Call function before main (mbed_main) |        1.43        |       20      |  1/1  | OK      | NUCLEO_F746ZG | GCC_ARM   | MBED_A9     | Serial Echo at 115200                 |        6.49        |       20      |  1/1 
OK      | NUCLEO_F746ZG | GCC_ARM   | MBED_BUSOUT | BusOut                                |        2.3         |       15      |  1/1 
OK      | NUCLEO_F746ZG | GCC_ARM   | RTOS_1      | Basic thread                          |       11.35        |       15      |  1/1 
OK      | NUCLEO_F746ZG | GCC_ARM   | RTOS_2      | Mutex resource lock                   |       11.39        |       20      |  1/1 
OK      | NUCLEO_F746ZG | GCC_ARM   | RTOS_3      | Semaphore resource lock               |        8.41        |       20      |  1/1 
OK      | NUCLEO_F746ZG | GCC_ARM   | RTOS_4      | Signals messaging                     |        6.37        |       20      |  1/1 
TIMEOUT | NUCLEO_F746ZG | GCC_ARM   | RTOS_5      | Queue messaging                       |       40.37        |       20      |  0/1 
TIMEOUT | NUCLEO_F746ZG | GCC_ARM   | RTOS_6      | Mail messaging                        |       40.36        |       20      |  0/1 
OK      | NUCLEO_F746ZG | GCC_ARM   | RTOS_7      | Timer                                 |       11.38        |       15      |  1/1 
OK      | NUCLEO_F746ZG | GCC_ARM   | RTOS_8      | ISR (Queue)                           |        6.35        |       20      |  1/1 

Result: 24 OK / 2 TIMEOUT
Completed in 379.24 sec


 Result  | Target       | Toolchain | Test ID     | Test Description                      | Elapsed Time (sec) | Timeout (sec) | Loops
---------|--------------|-----------|-------------|---------------------------------------|--------------------|---------------|-------
OK      | DISCO_F746NG | GCC_ARM   | DTCT_1      | Simple detect test                    |        0.5         |       10      |  1/1 
OK      | DISCO_F746NG | GCC_ARM   | EXAMPLE_1   | /dev/null                             |        3.45        |       20      |  1/1 
OK      | DISCO_F746NG | GCC_ARM   | MBED_10     | Hello World                           |        0.37        |       5       |  1/1 
OK      | DISCO_F746NG | GCC_ARM   | MBED_11     | Ticker Int                            |       11.34        |       15      |  1/1 
OK      | DISCO_F746NG | GCC_ARM   | MBED_12     | C++                                   |        1.37        |       10      |  1/1 
OK      | DISCO_F746NG | GCC_ARM   | MBED_16     | RTC                                   |        4.57        |       20      |  1/1 
OK      | DISCO_F746NG | GCC_ARM   | MBED_2      | stdio                                 |        0.74        |       20      |  1/1 
OK      | DISCO_F746NG | GCC_ARM   | MBED_23     | Ticker Int us                         |       11.37        |       15      |  1/1 
OK      | DISCO_F746NG | GCC_ARM   | MBED_24     | Timeout Int us                        |       11.39        |       15      |  1/1 
OK      | DISCO_F746NG | GCC_ARM   | MBED_25     | Time us                               |       11.36        |       15      |  1/1 
OK      | DISCO_F746NG | GCC_ARM   | MBED_26     | Integer constant division             |        1.38        |       20      |  1/1 
OK      | DISCO_F746NG | GCC_ARM   | MBED_34     | Ticker Two callbacks                  |       11.38        |       15      |  1/1 
OK      | DISCO_F746NG | GCC_ARM   | MBED_37     | Serial NC RX                          |       10.89        |       20      |  1/1 
OK      | DISCO_F746NG | GCC_ARM   | MBED_38     | Serial NC TX                          |       15.39        |       20      |  1/1 
OK      | DISCO_F746NG | GCC_ARM   | MBED_A1     | Basic                                 |        1.34        |       20      |  1/1 
OK      | DISCO_F746NG | GCC_ARM   | MBED_A21    | Call function before main (mbed_main) |        1.42        |       20      |  1/1 
OK      | DISCO_F746NG | GCC_ARM   | MBED_A9     | Serial Echo at 115200                 |        6.21        |       20      |  1/1 
OK      | DISCO_F746NG | GCC_ARM   | MBED_BUSOUT | BusOut                                |        2.27        |       15      |  1/1 
OK      | DISCO_F746NG | GCC_ARM   | RTOS_1      | Basic thread                          |       11.37        |       15      |  1/1 
OK      | DISCO_F746NG | GCC_ARM   | RTOS_2      | Mutex resource lock                   |       11.35        |       20      |  1/1 
OK      | DISCO_F746NG | GCC_ARM   | RTOS_3      | Semaphore resource lock               |        8.38        |       20      |  1/1 
OK      | DISCO_F746NG | GCC_ARM   | RTOS_4      | Signals messaging                     |        6.35        |       20      |  1/1 
TIMEOUT | DISCO_F746NG | GCC_ARM   | RTOS_5      | Queue messaging                       |       40.37        |       20      |  0/1 
TIMEOUT | DISCO_F746NG | GCC_ARM   | RTOS_6      | Mail messaging                        |       40.35        |       20      |  0/1 
OK      | DISCO_F746NG | GCC_ARM   | RTOS_7      | Timer                                 |       11.33        |       15      |  1/1 
OK      | DISCO_F746NG | GCC_ARM   | RTOS_8      | ISR (Queue)                           |        6.36        |       20      |  1/1 

Result: 24 OK / 2 TIMEOUT
Completed in 340.88 sec


 Result | Target        | Toolchain | Test ID     | Test Description                      | Elapsed Time (sec) | Timeout (sec) | Loops 
--------|---------------|-----------|-------------|---------------------------------------|--------------------|---------------|-------
OK     | NUCLEO_F767ZI | GCC_ARM   | DTCT_1      | Simple detect test                    |        0.53        |       10      |  1/1 
OK     | NUCLEO_F767ZI | GCC_ARM   | EXAMPLE_1   | /dev/null                             |        5.78        |       20      |  1/1 
OK     | NUCLEO_F767ZI | GCC_ARM   | MBED_10     | Hello World                           |        0.37        |       5       |  1/1 
OK     | NUCLEO_F767ZI | GCC_ARM   | MBED_11     | Ticker Int                            |       11.39        |       15      |  1/1 
OK     | NUCLEO_F767ZI | GCC_ARM   | MBED_12     | C++                                   |        1.41        |       10      |  1/1 
OK     | NUCLEO_F767ZI | GCC_ARM   | MBED_16     | RTC                                   |        4.57        |       20      |  1/1 
OK     | NUCLEO_F767ZI | GCC_ARM   | MBED_2      | stdio                                 |        0.79        |       20      |  1/1 
OK     | NUCLEO_F767ZI | GCC_ARM   | MBED_23     | Ticker Int us                         |       11.37        |       15      |  1/1 
OK     | NUCLEO_F767ZI | GCC_ARM   | MBED_24     | Timeout Int us                        |       11.41        |       15      |  1/1 
OK     | NUCLEO_F767ZI | GCC_ARM   | MBED_25     | Time us                               |        11.4        |       15      |  1/1 
OK     | NUCLEO_F767ZI | GCC_ARM   | MBED_26     | Integer constant division             |        1.41        |       20      |  1/1 
OK     | NUCLEO_F767ZI | GCC_ARM   | MBED_34     | Ticker Two callbacks                  |       11.41        |       15      |  1/1 
OK     | NUCLEO_F767ZI | GCC_ARM   | MBED_37     | Serial NC RX                          |        10.9        |       20      |  1/1 
OK     | NUCLEO_F767ZI | GCC_ARM   | MBED_38     | Serial NC TX                          |       15.38        |       20      |  1/1 
OK     | NUCLEO_F767ZI | GCC_ARM   | MBED_A1     | Basic                                 |        1.34        |       20      |  1/1 
OK     | NUCLEO_F767ZI | GCC_ARM   | MBED_A21    | Call function before main (mbed_main) |        1.45        |       20      |  1/1 
OK     | NUCLEO_F767ZI | GCC_ARM   | MBED_A9     | Serial Echo at 115200                 |        6.5         |       20      |  1/1 
OK     | NUCLEO_F767ZI | GCC_ARM   | MBED_BUSOUT | BusOut                                |        2.28        |       15      |  1/1 

Result: 18 OK
Completed in 174.85 sec

0xc0170 commented:
Travis fails - ``[WARNING] Using default settings. Define your settings in the file "workspace_tools/private_settings.py" or in "./mbed_settings.py"

Given MCU 'NUCLEOO_F746ZG' not into the supported list:``

0xc0170 commented:
cc @bcostm @adustm

ohagendorf commented:
I corrected the typos.

0xc0170 commented:
@bcostm @adustm Happy with this update?

bcostm commented:
Thanks @ohagendorf for what you've done.
It would be nice to have the result of automatic tests with ARM and IAR compilers. I'll try to do it this week.

ohagendorf commented:
Here are the test results with uARM config.
I'm very sorry but I don't have an IAR so I can submit these result.

Result | Target | Toolchain | Test ID | Test Description | Elapsed Time (sec) | Timeout (sec) | Loops |
--------|---------------|-----------|-------------|---------------------------------------|--------------------|---------------|-------|
 OK | NUCLEO_F746ZG | uARM | DTCT_1 | Simple detect test | 0.49 | 10 | 1/1 |
 OK | NUCLEO_F746ZG | uARM | EXAMPLE_1 | /dev/null | 3.46 | 20 | 1/1 |
 OK | NUCLEO_F746ZG | uARM | MBED_10 | Hello World | 0.37 | 5 | 1/1 |
 OK | NUCLEO_F746ZG | uARM | MBED_11 | Ticker Int | 11.36 | 15 | 1/1 |
 OK | NUCLEO_F746ZG | uARM | MBED_12 | C|| | 1.38 | 10 | 1/1 |
 OK | NUCLEO_F746ZG | uARM | MBED_16 | RTC | 4.73 | 20 | 1/1 |
 OK | NUCLEO_F746ZG | uARM | MBED_2 | stdio | 0.77 | 20 | 1/1 |
 OK | NUCLEO_F746ZG | uARM | MBED_23 | Ticker Int us | 11.36 | 15 | 1/1 |
 OK | NUCLEO_F746ZG | uARM | MBED_24 | Timeout Int us | 11.39 | 15 | 1/1 |
 OK | NUCLEO_F746ZG | uARM | MBED_25 | Time us | 11.36 | 15 | 1/1 |
 OK | NUCLEO_F746ZG | uARM | MBED_26 | Integer constant division | 1.37 | 20 | 1/1 |
 OK | NUCLEO_F746ZG | uARM | MBED_34 | Ticker Two callbacks | 11.37 | 15 | 1/1 |
 OK | NUCLEO_F746ZG | uARM | MBED_37 | Serial NC RX | 10.9 | 20 | 1/1 |
 OK | NUCLEO_F746ZG | uARM | MBED_38 | Serial NC TX | 15.4 | 20 | 1/1 |
 OK | NUCLEO_F746ZG | uARM | MBED_A1 | Basic | 1.34 | 20 | 1/1 |
 OK | NUCLEO_F746ZG | uARM | MBED_A21 | Call function before main (mbed_main) | 1.44 | 20 | 1/1 |
 OK | NUCLEO_F746ZG | uARM | MBED_A9 | Serial Echo at 115200 | 6.24 | 20 | 1/1 |
 OK | NUCLEO_F746ZG | uARM | MBED_BUSOUT | BusOut | 2.27 | 15 | 1/1 |
 OK | NUCLEO_F746ZG | uARM | RTOS_1 | Basic thread | 11.35 | 15 | 1/1 |
 OK | NUCLEO_F746ZG | uARM | RTOS_2 | Mutex resource lock | 11.37 | 20 | 1/1 |
 OK | NUCLEO_F746ZG | uARM | RTOS_3 | Semaphore resource lock | 8.38 | 20 | 1/1 |
 OK | NUCLEO_F746ZG | uARM | RTOS_4 | Signals messaging | 6.37 | 20 | 1/1 |
 OK | NUCLEO_F746ZG | uARM | RTOS_5 | Queue messaging | 2.4 | 20 | 1/1 |
 OK | NUCLEO_F746ZG | uARM | RTOS_6 | Mail messaging | 2.38 | 20 | 1/1 |
 OK | NUCLEO_F746ZG | uARM | RTOS_7 | Timer | 11.34 | 15 | 1/1 |
 OK | NUCLEO_F746ZG | uARM | RTOS_8 | ISR (Queue) | 6.37 | 20 | 1/1 |

Result: 26 OK


Result | Target | Toolchain | Test ID | Test Description | Elapsed Time (sec) | Timeout (sec) | Loops |
---------|---------------|-----------|-------------|---------------------------------------|--------------------|---------------|-------|
 OK | NUCLEO_F767ZI | uARM | DTCT_1 | Simple detect test | 0.5 | 10 | 1/1 |
 OK | NUCLEO_F767ZI | uARM | EXAMPLE_1 | /dev/null | 3.42 | 20 | 1/1 |
 OK | NUCLEO_F767ZI | uARM | MBED_10 | Hello World | 0.38 | 5 | 1/1 |
 OK | NUCLEO_F767ZI | uARM | MBED_11 | Ticker Int | 11.35 | 15 | 1/1 |
 TIMEOUT | NUCLEO_F767ZI | uARM | MBED_12 | C|| | 20.28 | 10 | 0/1 |
 OK | NUCLEO_F767ZI | uARM | MBED_16 | RTC | 4.58 | 20 | 1/1 |
 OK | NUCLEO_F767ZI | uARM | MBED_2 | stdio | 0.76 | 20 | 1/1 |
 OK | NUCLEO_F767ZI | uARM | MBED_23 | Ticker Int us | 11.37 | 15 | 1/1 |
 OK | NUCLEO_F767ZI | uARM | MBED_24 | Timeout Int us | 11.39 | 15 | 1/1 |
 OK | NUCLEO_F767ZI | uARM | MBED_25 | Time us | 11.37 | 15 | 1/1 |
 OK | NUCLEO_F767ZI | uARM | MBED_26 | Integer constant division | 1.37 | 20 | 1/1 |
 OK | NUCLEO_F767ZI | uARM | MBED_34 | Ticker Two callbacks | 11.38 | 15 | 1/1 |
 OK | NUCLEO_F767ZI | uARM | MBED_37 | Serial NC RX | 10.91 | 20 | 1/1 |
 OK | NUCLEO_F767ZI | uARM | MBED_38 | Serial NC TX | 15.41 | 20 | 1/1 |
 OK | NUCLEO_F767ZI | uARM | MBED_A1 | Basic | 1.37 | 20 | 1/1 |
 OK | NUCLEO_F767ZI | uARM | MBED_A21 | Call function before main (mbed_main) | 1.42 | 20 | 1/1 |
 OK | NUCLEO_F767ZI | uARM | MBED_A9 | Serial Echo at 115200 | 6.5 | 20 | 1/1 |
 OK | NUCLEO_F767ZI | uARM | MBED_BUSOUT | BusOut | 2.27 | 15 | 1/1 |

Result: 17 OK / 1 TIMEOUT


Result | Target | Toolchain | Test ID | Test Description | Elapsed Time (sec) | Timeout (sec) | Loops |
--------|--------------|-----------|-------------|---------------------------------------|--------------------|---------------|-------|
 OK | DISCO_F746NG | uARM | DTCT_1 | Simple detect test | 0.37 | 5 | 1/1 |
 OK | DISCO_F746NG | uARM | EXAMPLE_1 | /dev/null | 0.36 | 5 | 1/1 |
 OK | DISCO_F746NG | uARM | MBED_10 | Hello World | 0.37 | 5 | 1/1 |
 OK | DISCO_F746NG | uARM | MBED_11 | Ticker Int | 0.37 | 5 | 1/1 |
 OK | DISCO_F746NG | uARM | MBED_12 | C|| | 0.37 | 5 | 1/1 |
 OK | DISCO_F746NG | uARM | MBED_16 | RTC | 0.35 | 5 | 1/1 |
 OK | DISCO_F746NG | uARM | MBED_2 | stdio | 0.38 | 5 | 1/1 |
 OK | DISCO_F746NG | uARM | MBED_23 | Ticker Int us | 0.38 | 5 | 1/1 |
 OK | DISCO_F746NG | uARM | MBED_24 | Timeout Int us | 0.36 | 5 | 1/1 |
 OK | DISCO_F746NG | uARM | MBED_25 | Time us | 0.38 | 5 | 1/1 |
 OK | DISCO_F746NG | uARM | MBED_26 | Integer constant division | 0.37 | 5 | 1/1 |
 OK | DISCO_F746NG | uARM | MBED_34 | Ticker Two callbacks | 0.38 | 5 | 1/1 |
 OK | DISCO_F746NG | uARM | MBED_37 | Serial NC RX | 0.38 | 5 | 1/1 |
 OK | DISCO_F746NG | uARM | MBED_38 | Serial NC TX | 0.37 | 5 | 1/1 |
 OK | DISCO_F746NG | uARM | MBED_A1 | Basic | 0.38 | 5 | 1/1 |
 OK | DISCO_F746NG | uARM | MBED_A21 | Call function before main (mbed_main) | 0.38 | 5 | 1/1 |
 OK | DISCO_F746NG | uARM | MBED_A9 | Serial Echo at 115200 | 0.37 | 5 | 1/1 |
 OK | DISCO_F746NG | uARM | MBED_BUSOUT | BusOut | 0.38 | 5 | 1/1 |
 OK | DISCO_F746NG | uARM | RTOS_1 | Basic thread | 0.35 | 5 | 1/1 |
 OK | DISCO_F746NG | uARM | RTOS_2 | Mutex resource lock | 0.36 | 5 | 1/1 |
 OK | DISCO_F746NG | uARM | RTOS_3 | Semaphore resource lock | 0.35 | 5 | 1/1 |
 OK | DISCO_F746NG | uARM | RTOS_4 | Signals messaging | 0.37 | 5 | 1/1 |
 OK | DISCO_F746NG | uARM | RTOS_5 | Queue messaging | 0.38 | 5 | 1/1 |
 OK | DISCO_F746NG | uARM | RTOS_6 | Mail messaging | 0.35 | 5 | 1/1 |
 OK | DISCO_F746NG | uARM | RTOS_7 | Timer | 0.35 | 5 | 1/1 |
 OK | DISCO_F746NG | uARM | RTOS_8 | ISR (Queue) | 0.36 | 5 | 1/1 |

Result: 26 OK

ohagendorf commented:
Here are the results with ARM config and one of the F7 targets. Out of the box it is not possible to use this config (not only with this one but for all F7 anf F4 targets, maybe more) because of missing entries in startup_stm32f7xxx.s Another problem seems to be the stdio functionality. With several ```fflush(stdout);``` some more test are OK. But at the moment I didn't analyse it deeply. Maybe all failures are because the fflush problem.

 Result | Target | Toolchain | Test ID | Test Description | Elapsed Time (sec) | Timeout (sec) | Loops |
--------------|---------------|-----------|-------------|---------------------------------------|--------------------|---------------|-------|
 OK | NUCLEO_F746ZG | ARM | DTCT_1 | Simple detect test | 0.49 | 10 | 1/1 |
 TIMEOUT | NUCLEO_F746ZG | ARM | EXAMPLE_1 | /dev/null | 40.42 | 20 | 0/1 |
 OK | NUCLEO_F746ZG | ARM | MBED_10 | Hello World | 0.35 | 10 | 1/1 |
 TIMEOUT | NUCLEO_F746ZG | ARM | MBED_11 | Ticker Int | 30.38 | 15 | 0/1 |
 OK | NUCLEO_F746ZG | ARM | MBED_12 | C|| | 1.38 | 10 | 1/1 |
 TIMEOUT | NUCLEO_F746ZG | ARM | MBED_16 | RTC | 40.36 | 20 | 0/1 |
 OK | NUCLEO_F746ZG | ARM | MBED_2 | stdio | 5.75 | 20 | 1/1 |
 TIMEOUT | NUCLEO_F746ZG | ARM | MBED_23 | Ticker Int us | 30.38 | 15 | 0/1 |
 TIMEOUT | NUCLEO_F746ZG | ARM | MBED_24 | Timeout Int us | 30.38 | 15 | 0/1 |
 OK | NUCLEO_F746ZG | ARM | MBED_25 | Time us | 11.34 | 15 | 1/1 |
 OK | NUCLEO_F746ZG | ARM | MBED_26 | Integer constant division | 1.38 | 10 | 1/1 |
 TIMEOUT | NUCLEO_F746ZG | ARM | MBED_34 | Ticker Two callbacks | 30.38 | 15 | 0/1 |
 TIMEOUT | NUCLEO_F746ZG | ARM | MBED_37 | Serial NC RX | 40.37 | 20 | 0/1 |
 IOERR_SERIAL | NUCLEO_F746ZG | ARM | MBED_38 | Serial NC TX | 10.37 | 10 | 0/1 |
 OK | NUCLEO_F746ZG | ARM | MBED_A1 | Basic | 1.35 | 20 | 1/1 |
 OK | NUCLEO_F746ZG | ARM | MBED_A21 | Call function before main (mbed_main) | 1.44 | 20 | 1/1 |
 TIMEOUT | NUCLEO_F746ZG | ARM | MBED_A9 | Serial Echo at 115200 | 40.37 | 20 | 0/1 |
 TIMEOUT | NUCLEO_F746ZG | ARM | MBED_BUSOUT | BusOut | 30.27 | 15 | 0/1 |
 TIMEOUT | NUCLEO_F746ZG | ARM | RTOS_1 | Basic thread | 30.27 | 15 | 0/1 |
 TIMEOUT | NUCLEO_F746ZG | ARM | RTOS_2 | Mutex resource lock | 40.27 | 20 | 0/1 |
 TIMEOUT | NUCLEO_F746ZG | ARM | RTOS_3 | Semaphore resource lock | 40.25 | 20 | 0/1 |
 TIMEOUT | NUCLEO_F746ZG | ARM | RTOS_4 | Signals messaging | 20.27 | 10 | 0/1 |
 TIMEOUT | NUCLEO_F746ZG | ARM | RTOS_5 | Queue messaging | 20.27 | 10 | 0/1 |
 TIMEOUT | NUCLEO_F746ZG | ARM | RTOS_6 | Mail messaging | 20.27 | 10 | 0/1 |
 TIMEOUT | NUCLEO_F746ZG | ARM | RTOS_7 | Timer | 30.27 | 15 | 0/1 |
 TIMEOUT | NUCLEO_F746ZG | ARM | RTOS_8 | ISR (Queue) | 20.27 | 10 | 0/1 |

Result: 1 IOERR_SERIAL / 8 OK / 17 TIMEOUT

bcostm commented:
Thanks! I'll do IAR tests.

bcostm commented:
The issue with ARM compiler is due because now in the HAL_RCC_GetHCLKFreq() function the SystemCoreClock is no more updated... As a fix we need to call SystemCoreClockUpdate() before calling HAL_UART_Init(). We have also added the reset of the USART peripheral systematically in serial_init() function.

I did that and the result is OK:

```
Test summary:
+--------+---------------------+-----------+-------------+---------------------------------------+--------------------+---------------+-------+
| Result | Target              | Toolchain | Test ID     | Test Description                      | Elapsed Time (sec) | Timeout (sec) | Loops |
+--------+---------------------+-----------+-------------+---------------------------------------+--------------------+---------------+-------+
| OK     | NUCLEO_F746ZG[F928] | ARM       | DTCT_1      | Simple detect test                    |        0.52        |       10      |  1/1  |
| OK     | NUCLEO_F746ZG[F928] | ARM       | EXAMPLE_1   | /dev/null                             |        3.45        |       20      |  1/1  |
| OK     | NUCLEO_F746ZG[F928] | ARM       | MBED_10     | Hello World                           |        0.36        |       5       |  1/1  |
| OK     | NUCLEO_F746ZG[F928] | ARM       | MBED_11     | Ticker Int                            |        11.4        |       15      |  1/1  |
| OK     | NUCLEO_F746ZG[F928] | ARM       | MBED_12     | C++                                   |        1.39        |       10      |  1/1  |
| OK     | NUCLEO_F746ZG[F928] | ARM       | MBED_16     | RTC                                   |        4.59        |       20      |  1/1  |
| OK     | NUCLEO_F746ZG[F928] | ARM       | MBED_2      | stdio                                 |        0.7         |       20      |  1/1  |
| OK     | NUCLEO_F746ZG[F928] | ARM       | MBED_23     | Ticker Int us                         |       11.34        |       15      |  1/1  |
| OK     | NUCLEO_F746ZG[F928] | ARM       | MBED_24     | Timeout Int us                        |       11.43        |       15      |  1/1  |
| OK     | NUCLEO_F746ZG[F928] | ARM       | MBED_25     | Time us                               |       11.36        |       15      |  1/1  |
| OK     | NUCLEO_F746ZG[F928] | ARM       | MBED_26     | Integer constant division             |        1.42        |       20      |  1/1  |
| OK     | NUCLEO_F746ZG[F928] | ARM       | MBED_34     | Ticker Two callbacks                  |       11.36        |       15      |  1/1  |
| OK     | NUCLEO_F746ZG[F928] | ARM       | MBED_37     | Serial NC RX                          |        11.0        |       20      |  1/1  |
| OK     | NUCLEO_F746ZG[F928] | ARM       | MBED_38     | Serial NC TX                          |       15.48        |       20      |  1/1  |
| OK     | NUCLEO_F746ZG[F928] | ARM       | MBED_A1     | Basic                                 |        1.4         |       20      |  1/1  |
| OK     | NUCLEO_F746ZG[F928] | ARM       | MBED_A21    | Call function before main (mbed_main) |        1.44        |       20      |  1/1  |
| OK     | NUCLEO_F746ZG[F928] | ARM       | MBED_A9     | Serial Echo at 115200                 |        6.36        |       20      |  1/1  |
| OK     | NUCLEO_F746ZG[F928] | ARM       | MBED_BUSOUT | BusOut                                |        2.24        |       15      |  1/1  |
| OK     | NUCLEO_F746ZG[F928] | ARM       | RTOS_1      | Basic thread                          |       11.41        |       15      |  1/1  |
| OK     | NUCLEO_F746ZG[F928] | ARM       | RTOS_2      | Mutex resource lock                   |       11.43        |       20      |  1/1  |
| OK     | NUCLEO_F746ZG[F928] | ARM       | RTOS_3      | Semaphore resource lock               |        8.4         |       20      |  1/1  |
| OK     | NUCLEO_F746ZG[F928] | ARM       | RTOS_4      | Signals messaging                     |        6.41        |       20      |  1/1  |
| OK     | NUCLEO_F746ZG[F928] | ARM       | RTOS_5      | Queue messaging                       |        2.42        |       20      |  1/1  |
| OK     | NUCLEO_F746ZG[F928] | ARM       | RTOS_6      | Mail messaging                        |        2.41        |       20      |  1/1  |
| OK     | NUCLEO_F746ZG[F928] | ARM       | RTOS_7      | Timer                                 |       11.38        |       15      |  1/1  |
| OK     | NUCLEO_F746ZG[F928] | ARM       | RTOS_8      | ISR (Queue)                           |        6.38        |       20      |  1/1  |
+--------+---------------------+-----------+-------------+---------------------------------------+--------------------+---------------+-------+
Result: 26 OK
```

I'll try to send my changes in your PR.

bcostm commented:
Ok with uARM too:

```
Test summary:
+--------+---------------------+-----------+-------------+---------------------------------------+--------------------+---------------+-------+
| Result | Target              | Toolchain | Test ID     | Test Description                      | Elapsed Time (sec) | Timeout (sec) | Loops |
+--------+---------------------+-----------+-------------+---------------------------------------+--------------------+---------------+-------+
| OK     | NUCLEO_F746ZG[F928] | uARM      | DTCT_1      | Simple detect test                    |        0.5         |       10      |  1/1  |
| OK     | NUCLEO_F746ZG[F928] | uARM      | EXAMPLE_1   | /dev/null                             |        3.45        |       20      |  1/1  |
| OK     | NUCLEO_F746ZG[F928] | uARM      | MBED_10     | Hello World                           |        0.36        |       5       |  1/1  |
| OK     | NUCLEO_F746ZG[F928] | uARM      | MBED_11     | Ticker Int                            |       11.39        |       15      |  1/1  |
| OK     | NUCLEO_F746ZG[F928] | uARM      | MBED_12     | C++                                   |        1.4         |       10      |  1/1  |
| OK     | NUCLEO_F746ZG[F928] | uARM      | MBED_16     | RTC                                   |        4.6         |       20      |  1/1  |
| OK     | NUCLEO_F746ZG[F928] | uARM      | MBED_2      | stdio                                 |        0.72        |       20      |  1/1  |
| OK     | NUCLEO_F746ZG[F928] | uARM      | MBED_23     | Ticker Int us                         |        11.4        |       15      |  1/1  |
| OK     | NUCLEO_F746ZG[F928] | uARM      | MBED_24     | Timeout Int us                        |       11.42        |       15      |  1/1  |
| OK     | NUCLEO_F746ZG[F928] | uARM      | MBED_25     | Time us                               |       11.39        |       15      |  1/1  |
| OK     | NUCLEO_F746ZG[F928] | uARM      | MBED_26     | Integer constant division             |        1.39        |       20      |  1/1  |
| OK     | NUCLEO_F746ZG[F928] | uARM      | MBED_34     | Ticker Two callbacks                  |       11.37        |       15      |  1/1  |
| OK     | NUCLEO_F746ZG[F928] | uARM      | MBED_37     | Serial NC RX                          |       11.05        |       20      |  1/1  |
| OK     | NUCLEO_F746ZG[F928] | uARM      | MBED_38     | Serial NC TX                          |       15.57        |       20      |  1/1  |
| OK     | NUCLEO_F746ZG[F928] | uARM      | MBED_A1     | Basic                                 |        1.39        |       20      |  1/1  |
| OK     | NUCLEO_F746ZG[F928] | uARM      | MBED_A21    | Call function before main (mbed_main) |        1.44        |       20      |  1/1  |
| OK     | NUCLEO_F746ZG[F928] | uARM      | MBED_A9     | Serial Echo at 115200                 |        6.41        |       20      |  1/1  |
| OK     | NUCLEO_F746ZG[F928] | uARM      | MBED_BUSOUT | BusOut                                |        2.25        |       15      |  1/1  |
| OK     | NUCLEO_F746ZG[F928] | uARM      | RTOS_1      | Basic thread                          |       11.38        |       15      |  1/1  |
| OK     | NUCLEO_F746ZG[F928] | uARM      | RTOS_2      | Mutex resource lock                   |       11.53        |       20      |  1/1  |
| OK     | NUCLEO_F746ZG[F928] | uARM      | RTOS_3      | Semaphore resource lock               |        8.46        |       20      |  1/1  |
| OK     | NUCLEO_F746ZG[F928] | uARM      | RTOS_4      | Signals messaging                     |        6.47        |       20      |  1/1  |
| OK     | NUCLEO_F746ZG[F928] | uARM      | RTOS_5      | Queue messaging                       |        2.42        |       20      |  1/1  |
| OK     | NUCLEO_F746ZG[F928] | uARM      | RTOS_6      | Mail messaging                        |        2.42        |       20      |  1/1  |
| OK     | NUCLEO_F746ZG[F928] | uARM      | RTOS_7      | Timer                                 |       11.42        |       15      |  1/1  |
| OK     | NUCLEO_F746ZG[F928] | uARM      | RTOS_8      | ISR (Queue)                           |        6.47        |       20      |  1/1  |
+--------+---------------------+-----------+-------------+---------------------------------------+--------------------+---------------+-------+
Result: 26 OK
```

0xc0170 commented:
Thanks for the sync in here, as I understand , we wait for PR to be merged which should fix the issues mentioned above, let me know once completed

ohagendorf commented:
I rebased and added/tested ARM config. Tests are OK.
I can't say why I've got failures with MBED_37 and @bcostm got an OK.
I'll submit the STM32F767 mcu definition for progen a bit later.

Everything else is completed.

Result | Target | Toolchain | Test ID | Test Description | Elapsed Time (sec) | Timeout (sec) | Loops
-------------|---------------|-----------|-------------|---------------------------------------|--------------------|---------------|------
OK | NUCLEO_F746ZG | ARM | DTCT_1 | Simple detect test | 0.5 | 10 | 1/1 
OK | NUCLEO_F746ZG | ARM | EXAMPLE_1 | /dev/null | 3.45 | 20 | 1/1 
OK | NUCLEO_F746ZG | ARM | MBED_10 | Hello World | 0.37 | 5 | 1/1 
OK | NUCLEO_F746ZG | ARM | MBED_11 | Ticker Int | 11.36 | 15 | 1/1 
OK | NUCLEO_F746ZG | ARM | MBED_12 | C|| | 1.37 | 10 | 1/1 
OK | NUCLEO_F746ZG | ARM | MBED_16 | RTC | 4.61 | 20 | 1/1 
OK | NUCLEO_F746ZG | ARM | MBED_2 | stdio | 0.75 | 20 | 1/1 
OK | NUCLEO_F746ZG | ARM | MBED_23 | Ticker Int us | 11.38 | 15 | 1/1 
OK | NUCLEO_F746ZG | ARM | MBED_24 | Timeout Int us | 11.36 | 15 | 1/1 
OK | NUCLEO_F746ZG | ARM | MBED_25 | Time us | 11.37 | 15 | 1/1 
OK | NUCLEO_F746ZG | ARM | MBED_26 | Integer constant division | 1.38 | 20 | 1/1 
OK | NUCLEO_F746ZG | ARM | MBED_34 | Ticker Two callbacks | 11.37 | 15 | 1/1 
IOERR_SERIAL | NUCLEO_F746ZG | ARM | MBED_37 | Serial NC RX | 5.37 | 20 | 0/1 
OK | NUCLEO_F746ZG | ARM | MBED_38 | Serial NC TX | 10.4 | 20 | 1/1 
OK | NUCLEO_F746ZG | ARM | MBED_A1 | Basic | 1.35 | 20 | 1/1 
OK | NUCLEO_F746ZG | ARM | MBED_A21 | Call function before main (mbed_main) | 1.42 | 20 | 1/1 
OK | NUCLEO_F746ZG | ARM | MBED_A9 | Serial Echo at 115200 | 6.26 | 20 | 1/1 
OK | NUCLEO_F746ZG | ARM | MBED_BUSOUT | BusOut | 2.27 | 15 | 1/1 
OK | NUCLEO_F746ZG | ARM | RTOS_1 | Basic thread | 11.35 | 15 | 1/1 
OK | NUCLEO_F746ZG | ARM | RTOS_2 | Mutex resource lock | 11.37 | 20 | 1/1 
OK | NUCLEO_F746ZG | ARM | RTOS_3 | Semaphore resource lock | 8.35 | 20 | 1/1 
OK | NUCLEO_F746ZG | ARM | RTOS_4 | Signals messaging | 6.37 | 20 | 1/1 
OK | NUCLEO_F746ZG | ARM | RTOS_5 | Queue messaging | 2.37 | 20 | 1/1 
OK | NUCLEO_F746ZG | ARM | RTOS_6 | Mail messaging | 2.37 | 20 | 1/1 
OK | NUCLEO_F746ZG | ARM | RTOS_7 | Timer | 11.36 | 15 | 1/1 
OK | NUCLEO_F746ZG | ARM | RTOS_8 | ISR (Queue) | 6.37 | 20 | 1/1 



Result | Target | Toolchain | Test ID | Test Description | Elapsed Time (sec) | Timeout (sec) | Loops
-------------|--------------|-----------|-------------|---------------------------------------|--------------------|---------------|------
OK | DISCO_F746NG | ARM | DTCT_1 | Simple detect test | 0.49 | 10 | 1/1 
OK | DISCO_F746NG | ARM | EXAMPLE_1 | /dev/null | 3.42 | 20 | 1/1 
OK | DISCO_F746NG | ARM | MBED_10 | Hello World | 0.37 | 5 | 1/1 
OK | DISCO_F746NG | ARM | MBED_11 | Ticker Int | 11.34 | 15 | 1/1 
OK | DISCO_F746NG | ARM | MBED_12 | C|| | 1.35 | 10 | 1/1 
OK | DISCO_F746NG | ARM | MBED_16 | RTC | 4.59 | 20 | 1/1 
OK | DISCO_F746NG | ARM | MBED_2 | stdio | 0.73 | 20 | 1/1 
OK | DISCO_F746NG | ARM | MBED_23 | Ticker Int us | 11.37 | 15 | 1/1 
OK | DISCO_F746NG | ARM | MBED_24 | Timeout Int us | 11.37 | 15 | 1/1 
OK | DISCO_F746NG | ARM | MBED_25 | Time us | 11.36 | 15 | 1/1 
OK | DISCO_F746NG | ARM | MBED_26 | Integer constant division | 1.38 | 20 | 1/1 
OK | DISCO_F746NG | ARM | MBED_34 | Ticker Two callbacks | 11.35 | 15 | 1/1 
IOERR_SERIAL | DISCO_F746NG | ARM | MBED_37 | Serial NC RX | 5.37 | 20 | 0/1 
OK | DISCO_F746NG | ARM | MBED_38 | Serial NC TX | 10.38 | 20 | 1/1 
OK | DISCO_F746NG | ARM | MBED_A1 | Basic | 1.36 | 20 | 1/1 
OK | DISCO_F746NG | ARM | MBED_A21 | Call function before main (mbed_main) | 1.43 | 20 | 1/1 
OK | DISCO_F746NG | ARM | MBED_A9 | Serial Echo at 115200 | 6.26 | 20 | 1/1 
OK | DISCO_F746NG | ARM | MBED_BUSOUT | BusOut | 2.27 | 15 | 1/1 
OK | DISCO_F746NG | ARM | RTOS_1 | Basic thread | 11.34 | 15 | 1/1 
OK | DISCO_F746NG | ARM | RTOS_2 | Mutex resource lock | 11.37 | 20 | 1/1 
OK | DISCO_F746NG | ARM | RTOS_3 | Semaphore resource lock | 8.38 | 20 | 1/1 
OK | DISCO_F746NG | ARM | RTOS_4 | Signals messaging | 6.36 | 20 | 1/1 
OK | DISCO_F746NG | ARM | RTOS_5 | Queue messaging | 2.38 | 20 | 1/1 
OK | DISCO_F746NG | ARM | RTOS_6 | Mail messaging | 2.37 | 20 | 1/1 
OK | DISCO_F746NG | ARM | RTOS_7 | Timer | 11.35 | 15 | 1/1 
OK | DISCO_F746NG | ARM | RTOS_8 | ISR (Queue) | 6.36 | 20 | 1/1 



Result | Target | Toolchain | Test ID | Test Description | Elapsed Time (sec) | Timeout (sec) | Loops
-------|---------------|-----------|-------------|---------------------------------------|--------------------|---------------|------
OK | NUCLEO_F767ZI | ARM | DTCT_1 | Simple detect test | 2.27 | 10 | 1/1 
OK | NUCLEO_F767ZI | ARM | EXAMPLE_1 | /dev/null | 2.27 | 10 | 1/1 
OK | NUCLEO_F767ZI | ARM | MBED_10 | Hello World | 2.27 | 10 | 1/1 
OK | NUCLEO_F767ZI | ARM | MBED_11 | Ticker Int | 2.27 | 20 | 1/1 
OK | NUCLEO_F767ZI | ARM | MBED_12 | C|| | 2.27 | 10 | 1/1 
OK | NUCLEO_F767ZI | ARM | MBED_16 | RTC | 2.27 | 15 | 1/1 
OK | NUCLEO_F767ZI | ARM | MBED_2 | stdio | 2.27 | 20 | 1/1 
OK | NUCLEO_F767ZI | ARM | MBED_23 | Ticker Int us | 2.27 | 15 | 1/1 
OK | NUCLEO_F767ZI | ARM | MBED_24 | Timeout Int us | 2.27 | 15 | 1/1 
OK | NUCLEO_F767ZI | ARM | MBED_25 | Time us | 2.28 | 15 | 1/1 
OK | NUCLEO_F767ZI | ARM | MBED_26 | Integer constant division | 2.28 | 10 | 1/1 
OK | NUCLEO_F767ZI | ARM | MBED_34 | Ticker Two callbacks | 2.29 | 15 | 1/1 
IOERR_SERIAL | NUCLEO_F767ZI | ARM | MBED_37 | Serial NC RX | 2.27 | 10 | 1/1 
OK | NUCLEO_F767ZI | ARM | MBED_38 | Serial NC TX | 2.27 | 10 | 1/1 
OK | NUCLEO_F767ZI | ARM | MBED_A1 | Basic | 2.29 | 10 | 1/1 
OK | NUCLEO_F767ZI | ARM | MBED_A21 | Call function before main (mbed_main) | 2.27 | 10 | 1/1 
OK | NUCLEO_F767ZI | ARM | MBED_A9 | Serial Echo at 115200 | 2.29 | 10 | 1/1 
OK | NUCLEO_F767ZI | ARM | MBED_BUSOUT | BusOut | 2.27 | 15 | 1/1

0xc0170 commented:
Not sure what has happened, but I see here commits from master?

@bcostm happy with it?

ohagendorf commented:
It was nothing special: I rebased it and (after that) merged the PR ohagendorf/mbed#3

bcostm commented:
Yes ok with the Nucleo_F746ZG.

For the new Nucleo_F767 I didn't test it but it looks ok too. Note that this board is not planned to be on mbed IDE for the moment... But thanks again @ohagendorf

ohagendorf commented:
I want to use this with my students in my own environment, a Matlab/Simulink - mbed target. We are using only the offline toolchain.

When everybody is satisfied with this and it is merged I can also prepare #1823

ohagendorf commented:
I'm closing this because of the strange rebase result. Please see #1909. It is identical.
